prefect_gcp/cloud_run.py

```diff
@@ -135,7 +136,15 @@ def create(client: Resource, namespace: str, body: dict):
     def delete(client: Resource, namespace: str, job_name: str):
         """Make a delete request to the GCP jobs API."""
         request = client.jobs().delete(name=f"namespaces/{namespace}/jobs/{job_name}")
-        response = request.execute()
+        try:
+            response = request.execute()
+        except Exception as exc:
+            if "does not exist" in str(exc):
+                raise InfrastructureNotFound(
```

Comment:
I'm wondering if this would be better to catch inside of the `kill` method in the infrastructure block. In theory `Job` is an independent class that doesn't have knowledge of `CloudRunJob` and its associated logic.